plugins/CorePluginsAdmin/vue/src/Field/Field.adapter.ts

```diff
@@ -0,0 +1,211 @@
+    return transformAngularJsBoolAttr(value);
+  }
+
+  return value;
+}
+
+export default createAngularJsAdapter<[ITimeoutService]>({
+  component: Field,
+  require: '?ngModel',
+  scope: {
+    uicontrol: {
+      angularJsBind: '@',
+    },
+    name: {
+      angularJsBind: '@',
+    },
+    value: {
+      vue: 'modelValue',
+      angularJsBind: '@',
+      transform(value, vm, scope) {
+        // vue components expect object data as input, so we parse JSON data
+        // for angularjs directives that use JSON.
+        return handleJsonValue(value, scope.varType, scope.uicontrol);
+      },
+    },
+    default: {
+      vue: 'defaultValue',
+      angularJsBind: '@',
+    },
+    options: {
+      angularJsBind: '=',
+    },
+    description: {
+      angularJsBind: '@',
+    },
+    introduction: {
+      angularJsBind: '@',
+    },
+    title: {
+      angularJsBind: '@',
+    },
+    inlineHelp: {
+      angularJsBind: '@',
+    },
+    disabled: {
+      angularJsBind: '=',
+      transform: transformAngularJsBoolAttr,
+    },
+    uiControlAttributes: {
+      angularJsBind: '=',
+    },
+    uiControlOptions: {
+      angularJsBind: '=',
+    },
+    autocomplete: {
+      angularJsBind: '@',
+      transform: transformAngularJsBoolAttr,
+    },
+    condition: {
+      angularJsBind: '@',
+      transform(value, vm, scope) {
+        let transformed = value;
+        if (value) {
+          transformed = (values: unknown[]) => scope.$eval(value, values);
+        }
+        return transformed;
+      },
+    },
+    varType: {
+      angularJsBind: '@',
+    },
+    autofocus: {
+      angularJsBind: '@',
+      transform: transformAngularJsBoolAttr,
+    },
+    tabindex: {
+      angularJsBind: '@',
+      transform: transformAngularJsIntAttr,
+    },
+    fullWidth: {
+      angularJsBind: '@',
+      transform: transformAngularJsBoolAttr,
+    },
+    maxlength: {
+      angularJsBind: '@',
+      transform: transformAngularJsIntAttr,
+    },
+    required: {
+      angularJsBind: '@',
+      transform: transformAngularJsBoolAttr,
+    },
+    placeholder: {
+      angularJsBind: '@',
+    },
+    rows: {
+      angularJsBind: '@',
+      transform: transformAngularJsIntAttr,
+    },
+    min: {
+      angularJsBind: '@',
+      transform: transformAngularJsIntAttr,
+    },
+    max: {
+      angularJsBind: '@',
+      transform: transformAngularJsIntAttr,
+    },
+    component: {
+      angularJsBind: '<',
+      transform(value, vm, scope) {
+        if (!value) {
+          return value;
+        }
+
+        if (scope.templateFile) {
+          return shallowRef(FieldAngularJsTemplate);
+        }
+
+        const { plugin, component } = value;
+        if (!plugin || !component) {
+          throw new Error("Invalid component property given to piwik-field directive, must be {plugin: '...',component: '...'}");
+        }
+
+        return shallowRef(useExternalPluginComponent(plugin, component));
+      },
```

Comment:
Converts the angularjs `component` property which will be a JSON object to a Vue component. From Vue, we'll just pass a component directly.